bootstrap/internal/locking/control_plane_init_mutex.go

```diff
@@ -0,0 +1,191 @@
+/*
```

Comment:
Haven't gone into depth on this file, looks like a copy paste from upstream?

Reply:
Yes, it's internal upstream, so had to copy it over